src/extractor/restriction_parser.cpp

```diff
@@ -160,7 +173,42 @@ RestrictionParser::TryParse(const osmium::Relation &relation) const
             break;
         }
     }
-    return boost::make_optional(std::move(restriction_container));
+
+    // parse conditional tags
+    if (parse_conditionals)
+    {
+        osmium::tags::KeyFilter::iterator fi_begin(filter, tag_list.begin(), tag_list.end());
+        osmium::tags::KeyFilter::iterator fi_end(filter, tag_list.end(), tag_list.end());
+        for (; fi_begin != fi_end; ++fi_begin)
+        {
+            const std::string key(fi_begin->key());
+            const std::string value(fi_begin->value());
+            // Parse condition and add independent value/condition pairs
+            const auto &parsed = osrm::util::ParseConditionalRestrictions(value);
+
+            if (parsed.empty())
+            {
+                osrm::util::Log(logWARNING) << "Conditional restriction parsing failed for \""
+                                            << fi_begin->value() << "\" at the turn "
+                                            << restriction_container.restriction.from.way << " -> "
+                                            << restriction_container.restriction.via.node << " -> "
+                                            << restriction_container.restriction.to.way;
+                continue;
+            }
+
+            for (auto &p : parsed)
+            {
+                std::vector<util::OpeningHours> parsed = util::ParseOpeningHours(p.condition);
+                if (!parsed.empty())
+                    restriction_container.restriction.condition = std::move(parsed);
+            }
+        }
+    }
+
+    // push back a copy of turn restriction
+    parsed_restrictions.push_back(boost::make_optional(restriction_container));
+
+    return std::move(parsed_restrictions);
```

Comment:
The C++ guarantees for function-local variables to be moved out by default. If you add a move you penalize further optimization.

By which I mean

    T fn() {
      T tmp;
      ..
      return tmp; // will be moved out if possible
    }

`return move(tmp)` would be bad for optimization here. Also there are situations in which you have to move:


    struct S {
      T gn() {
        return tmp; // will _not_ move tmp out, as tmp is not a function-local variable
      }

      T tmp;
    };

(will crash if you cann gn() more than once, since a moved-from object is in a valid but unspecified state)

There are some articles out there, search for penalizing move, RVO, e.g.

https://www.ibm.com/developerworks/community/blogs/5894415f-be62-4bc0-81c5-3956e82276f3/entry/RVO_V_S_std_move?lang=en